From 2b18daa64c37437d3601b090648a991c291e7c69 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Maxime=20M=C3=A9r=C3=A9?= <maxime.mere@foss.st.com>
Date: Tue, 1 Jul 2025 09:42:24 +0200
Subject: [PATCH 884/954] fix(stm32mp2): increase stack size when Mbed TLS is
 used
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

In the M33TD flavor, when trusted boot is enabled, a stack overflow
occurs when LTO is enabled. Increase the stack size to 4 KB in this
situation.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 56e9462512e604f20dc68ee5247418a11e3f96df fix(stm32mp2): increase stack size when Mbed TLS is used]
Change-Id: I50d65e8073073f30ad662ec6fc3690909888f6c2
Signed-off-by: Maxime Méré <maxime.mere@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/471245
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
---
 plat/st/stm32mp2/include/platform_def.h | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/plat/st/stm32mp2/include/platform_def.h b/plat/st/stm32mp2/include/platform_def.h
index 3ce8d40240..6686e89932 100644
--- a/plat/st/stm32mp2/include/platform_def.h
+++ b/plat/st/stm32mp2/include/platform_def.h
@@ -19,7 +19,12 @@
  ******************************************************************************/
 
 /* Size of cacheable stacks */
+#if STM32MP_CRYPTO_USE_SW
+/* mbedtls needs more stack, especially with LTO */
+#define PLATFORM_STACK_SIZE		0x1000
+#else
 #define PLATFORM_STACK_SIZE		0xC00
+#endif
 
 #define STM32MP_PRIMARY_CPU		U(0x0)
 #define STM32MP_SECONDARY_CPU		U(0x1)
-- 
2.53.0

